### PR TITLE
Use BuildKit with all Docker build commands for cache

### PR DESCRIPTION
**Pull request by humphd**
I want to see if I can force `docker-compose` to use BuildKit and the caching behaviour we want for our image builds.  This might not work, I need to test in CI.

### CODE REVIEW

**Comment by humphd** [comment]
I've got this working finally.  Here's what it does:

- I've rewritten `.github/workflows/e2e-tests-ci.yml`.  The two most significant changes are that I've limited the Docker containers we start to only those needed to pass the tests (see the `DOCKER_CONTAINERS` environment variable).  Second, I've switched to use `docker compose` (v2) vs. `docker-compose` (v1), since the latter isn't compatible with Docker cache, which we want to use.  The newer `docker compose` uses `docker` vs. python scripts to process the `docker-compose.yml` files.
- I've added a `pnpm:services pull` command, so that we can `pull` images from our registry if we don't want to build.
- I've updated the docs

You can see what this looks like when it runs in this workflow run:

https://github.com/Seneca-CDOT/telescope/runs/5604170384?check_suite_focus=true

This will cut the total running time by more than 50% for the e2e tests, from ~13-18m minutes down to ~7m

NOTE: the action I've re-written will not get run until we merge this (i.e., it will run the current one on `master`).

**Comment by humphd** [comment]
Fixes #3230

**Review by JiaHua-Zou — APPROVED** [review]

**Review by menghif — APPROVED** [review]
I love the new total duration in CI!

File: .github/workflows/e2e-tests-ci.yml [hunk 1]
```diff
@@ -10,22 +10,33 @@ jobs:
     strategy:
       matrix:
         node-version: [14.x, 16.x]
+    env:
+      # Whichever Docker containers are needed to pass e2e tests, define them here
+      DOCKER_CONTAINERS: 'sso feed-discovery auth login kong rest meta'
```
Comment (rclee91) [inline .github/workflows/e2e-tests-ci.yml:15]: I am just wondering if we want to add and test Parser e2e tests, if this means we'll have to add more images into this line here?

Comment (humphd) [reply .github/workflows/e2e-tests-ci.yml:15]: Correct.

**Pull request merged**